module/VuFind/src/VuFind/Related/Bookplate.php

```diff
@@ -0,0 +1,243 @@
+        $this->bookplateStrs = $this->getBookplateData('titles');
+        $this->bookplateImages = $this->getBookplateData('imgFull');
+        $this->bookplateThumbnails = $this->getBookplateData('imgThumb');
+        $this->fullUrlTemplate = $this->getBookplateFullUrlTemplate();
+        $this->thumbUrlTemplate = $this->getBookplateThumbUrlTemplate();
+        $this->displayTitles = $this->displayBookplateTitles();
+    }
+
+    /**
+     * Get an array of data representing bookplates.
+     *
+     * @param $s string name of data to retrieve.
+     *
+     * @return array
+     */
+    protected function getBookplateData($s)
+    {
+        $data = [
+            'titles' => $this->getBookplateTitlesField(),
+            'imgThumb' => $this->getBookplateThumbnailsField(),
+            'imgFull' => $this->getBookplateFullImagesField(),
+        ];
+        $field = $data[$s];
+        if (!empty($field) && isset($this->fields[$field])) {
+            return $this->fields[$field];
+        }
+        return [];
+    }
+
+    /**
+     * Get the full bookplate URL string template.
+     *
+     * @return string
+     */
+    protected function getBookplateFullUrlTemplate()
+    {
+        return $this->config->bookplate_full ?? '';
+    }
+
+    /**
+     * Get the bookplate URL thumbnail string template.
+     *
+     * @return string
+     */
+    protected function getBookplateThumbUrlTemplate()
+    {
+        return $this->config->bookplate_thumb ?? '';
+    }
+
+    /**
+     * Display titles under bookplates.
+     *
+     * @return boolean
+     */
+    protected function displayBookplateTitles()
+    {
+        return $this->config->bookplate_display_title ?? true;
+    }
+
+    /**
+     * Get a data field with an array of bookplate image titles.
+     *
+     * @return string
+     */
+    protected function getBookplateTitlesField()
+    {
+        return $this->config->bookplate_titles_field ?? '';
+    }
+
+    /**
+     * Get a data field with an array of strings that represent full images.
+     * These could be the unique parts of image names (e.g. donor code) or
+     * full paths to image files.
+     *
+     * @return string
+     */
+    protected function getBookplateFullImagesField()
+    {
+        return $this->config->bookplate_images_field ?? '';
+    }
+
+    /**
+     * Get a data field with an array of strings that represent thumbnails.
+     * These could be the unique parts of thumbnail names (e.g. donor code)
+     * or full paths to thumbnail image files.
+     *
+     * @return string
+     */
+    protected function getBookplateThumbnailsField()
+    {
+        return $this->config->bookplate_thumbnails_field ?? '';
+    }
+
+    /**
+     * Get bookplate details for display.
+     *
+     * @return array
+     */
+    public function getBookplateDetails()
+    {
+        $sameLen = count($this->bookplateStrs) == count($this->bookplateImages);
+        $hasBookplates = !empty($this->bookplateStrs)
```

Comment:
Technically, you don't need to check that both bookplateStrs and bookplateImages are non-empty if you're also confirming that they're the same length -- you can just check one and safely assume that they're both the same.

Reply:
Yes, this also no longer makes sense since we allow any combination of thumbnails, full-size images and titles. I had meant to change this and apparently forgot. Since the string is required for alt text, I think it makes sense to anchor to that. That's what I've done.

Reply:
Right, that makes sense! The only thing to be careful of is that now, if you do have inconsistent numbers of values in the fields, you can no longer assume that `$this->bookplateImages[$i]` and `$this->bookplateThumbnails[$i]` will be set inside the loop, so you should probably add some default values to avoid PHP notices -- e.g. add `?? null` or `?? ''` (depending on what default value makes the most sense).

Reply:
I set this up with `?? ''`. I think that makes sense. I tested it to confirm I could force a PHP notice with bad data before and confirmed that this fixed it (it did).

Reply:
Great, thanks!